src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -2710,6 +2874,40 @@ static class C
         CompileAndVerify(comp, expectedOutput: "aa rSystem.Object[] ra ra").VerifyDiagnostics();
     }
 
+    [Fact]
+    public void OverloadResolution_ReadOnlySpanVsArray_05()
+    {
+        var source = """
+            using System;
+
+            C.M(null);
+            C.M(default);
+
+            static class C
+            {
+                public static void M(object[] x) => Console.Write(1);
+                public static void M(ReadOnlySpan<object> x) => Console.Write(2);
+            }
+            """;
+        var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12);
+        CompileAndVerify(comp, expectedOutput: "11").VerifyDiagnostics();
+
+        // PROTOTYPE: Need to consider "implicit span conversion" in "better conversion target" for this to work.
+
+        var expectedDiagnostics = new[]
+        {
+            // (3,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(object[])' and 'C.M(ReadOnlySpan<object>)'
+            // C.M(null);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(object[])", "C.M(System.ReadOnlySpan<object>)").WithLocation(3, 3),
```

Comment:
I'm not sure if this should be an ambiguity - the problem is that `null` to Span UDC is not ignored and we don't have an equivalent standard conversion defined. So UDC is happening here instead of implicit span conversion which would be preferred (and there would be no ambiguity). Should the spec be updated with `null` to Span conversions?

If not then when we make the conversions "from type" ,"better conversion target" will kick in and the array overload will be picked.

Reply:
That's true. However, consider `M(ReadOnlySpan<int>)` called as `M(null)` will use a user-defined conversion operator defined on ReadOnlySpan (from `int[]`) - so there is a "hole" in that we do not handle this case via standard implicit span conversions. Is that fine?

Reply:
> so there is a "hole" in that we do not handle this case via standard implicit span conversions. Is that fine?

That seems fine to me. We don't stack implicit standard conversions, and `null` being allowed to convert to `Span` via the UDC is often considered not great.

Reply:
> and `null` being allowed to convert to `Span` via the UDC is often considered not great.

Just to be clear - that will continue to be allowed.

Reply:
Right, but it's ok to me to not handle it through stacked standard implicit conversions.